### PR TITLE
PostCSS as preprocessor

### PR DESCRIPTION
**Pull request by sveisvei**
### Breaking changes:
* Package now only exports main files and browser css files, (exception is thirdparty legacy folder)

### Bugfix:
* fix gulp task minifying to support legacy IE declaration hacks, i.e. "*_" prefixing.

### Added:
* Added script for "auto formatting/cleanup" source files (format output in later commit)
* imageoptim of all images as part of the `package` gulp task.
* postcss support via `postcss.config.js`
    * imports
    * variables
    * media minmax
    * custom media queries
    * autoprefixer

### Removed:
* server.js and readme entry for the dummy server.
* removed blatant copy of `src/styles` folder

### Followup
* add/rename more media queries https://github.com/finn-no/spaden/issues/30
* add more color variables, a lot of colors still defined inline https://github.com/finn-no/spaden/issues/31
* update readme https://github.com/finn-no/spaden/issues/29

(will add a following PR with the output from the `format-stylesheet` formatter.)

@hzr @antidecaf feedback welcome.

### CODE REVIEW

**Comment by hzr** [comment]
Why is server.js removed? Not sure if it works with all these changes, but something like that is very nice when developing. Unless it can be achieved in some other way now?

File: src/styles/browsers/ie8.css [hunk 1]
```diff
@@ -1,125 +1,125 @@
...
-	padding-bottom: 5px;
-}
\ No newline at end of file
+.container { max-width: 1000px; margin: 0 auto; }
+.container.kindawide { max-width: 1240px; }
+.page { width: 990px; }
+.page.fullsize{ width:100%; } /* Fix for jobb, på cv vinner 990px og ting havner i overflow  */
+
+/* Spacing */
+h1, h2, h3, h4, h5, h6, ul, ol, dl, p,blockquote, .media {margin:10px 20px;}
```
Comment (hzr) [inline src/styles/browsers/ie8.css]: This is not formatted, should it be?

File: src/styles/components/photoframe/legacy_crap_photoframe.css [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 .photoframe img { display: block; }
 
-/* eksperimentel kode for � beskj�re thumbs til samme st�rrelse */
+/* eksperimentel kode for � beskj�re thumbs til samme st�rrelse */
```
Comment (hzr) [inline src/styles/components/photoframe/legacy_crap_photoframe.css]: Encoding error.

File: src/styles/components/ribbons/ribbons.css [hunk 3]
```diff
@@ -15,10 +15,7 @@
     transform:rotate(45deg);
     text-align: center;
     background: #fff264;
-    background-image: -moz-linear-gradient(right, #fff264, #ffffe3);
-    background-image: -webkit-linear-gradient(right, #fff264, #ffffe3);
-    background-image: -o-linear-gradient(right, #fff264, #ffffe3);
-    background-image: linear-gradient(right, #fff264, #ffffe3);
+    background-image: linear-gradient(to left, #fff264, #ffffe3);
```
Comment (hzr) [inline src/styles/components/ribbons/ribbons.css]: Out of curiosity, did some postcss plugin insist on "to left" instead of "right"?

File: src/styles/config/variables.css [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    --pale-gray: #f5f5f5;
+    --dkdk-blue: #000066;
+    --dk-blue: #0099ff;
+    --lt-blue: #ddeeff;
+    --pale-blue: #f1f8ff;
+    --contrast: #444;
+    --firm: #ff6600;
+    --snotgreen: #d2e286;
+    --pissyellow: #ffff99;
+
```
Comment (hzr) [inline src/styles/config/variables.css]: Extra line break, also on line 70.

File: src/styles/config/variables.css [hunk 5]
```diff
@@ -0,0 +1,71 @@
+@custom-media --mini-screen (max-width: 459px);
+@custom-media --not-mini-screen (min-width: 460px);
+@custom-media --small-but-not-mini (min-width: 460px) and (max-width: 767px);
```
Comment (hzr) [inline src/styles/config/variables.css]: This shouldn't be necessary, we can just do e.g. `@media (--not-mini-screen) and (--small-screen)`.

File: src/styles/config/variables.css [hunk 6]
```diff
@@ -0,0 +1,71 @@
+@custom-media --mini-screen (max-width: 459px);
+@custom-media --not-mini-screen (min-width: 460px);
```
Comment (hzr) [inline src/styles/config/variables.css]: I thought this would be possible with `@media not (--mini-screen)`, but no browser seems to e.g.

```
@media not (min-width: 460px)
```

However, browser support this

```
@media not all and (min-width: 460px)
```

so we this can probably be rewritten as `@custom-media all and --mini-screen (max-width: 459px);` and then it can be negated when used.

File: README.md [hunk 7]
```diff
@@ -42,15 +42,6 @@ When releasing you must specify how to bump the version number. Please consult t
...
 
-## Local development
-
-	$ npm start
-
-```html
-<link rel="stylesheet" href="http://localhost:3000/">
-```
-
-This includes core.css and components.css
```
Comment (runeh) [inline README.md]: Should probably add some docs about what postcss plugins are enabled, and what features the provide.

File: src/styles/browsers/ie8.css (hunk 1 above)
Comment (sveisvei) [inline src/styles/browsers/ie8.css]: Formatting will be own PR.

File: src/styles/components/ribbons/ribbons.css (hunk 3 above)
Comment (sveisvei) [inline src/styles/components/ribbons/ribbons.css]: This is the latest spec.(https://developer.mozilla.org/en-US/docs/Web/CSS/linear-gradient)

File: src/styles/config/variables.css (hunk 4 above)
Comment (sveisvei) [inline src/styles/config/variables.css]: Doing formatting of all files in own PR.

File: src/styles/config/variables.css (hunk 5 above)
Comment (sveisvei) [inline src/styles/config/variables.css]: nesting of custom-media did not work out of the box. We can fix it maybe, not sure about the spec.

File: README.md (hunk 7 above)
Comment (sveisvei) [inline README.md]: Agree, @hzr will take it https://github.com/finn-no/spaden/issues/29

File: src/styles/core/grid/grids.css [hunk 8]
```diff
@@ -186,27 +194,29 @@ html {
...
 }
 
-@media screen and (max-width:459px){
+@media (--mini-screen){
     .r-cols2to6.bordered > .unit:nth-child(2n),
-    .r-cols2to5.bordered > .unit:nth-child(2n) { border-right:0;}
+    .r-cols2to5.bordered > .unit:nth-child(2n) { border-right: 0;}
     .r-cols2to6.bordered > .unit:nth-child(-n+2),
-    .r-cols2to5.bordered > .unit:nth-child(-n+2){border-top:0;}
+    .r-cols2to5.bordered > .unit:nth-child(-n+2){ border-top: 0;}
```
Comment (antidecaf) [inline src/styles/core/grid/grids.css]: Is this the autoformatter in action? Seems to kind of inconsistently put spaces inside brackets (comparing line 216 to 236 for example)

Comment (hzr) [inline src/styles/core/grid/grids.css]: See https://github.com/finn-no/spaden/pull/28#discussion_r31508498
